src/Memory.php

```diff
@@ -148,8 +149,13 @@ public function getIterator()
      */
     public function flush()
     {
+        if ($this->data === []) {
+            return true;
+        }
+
         $this->data = [];
-        return true;
+
+        return gc_mem_caches() >= 0;
```

Comment:
`gc_mem_caches` — Reclaims memory used by the Zend Engine memory manager

Reply:
`gc_mem_caches()`  needs to be called after `$this->data` is reset. I only added the `>= 0` because that method is supposed to return a `bool`.

Reply:
`gc_mem_caches() > 0` would return bool aswell.
Why is it okay that it returns `0 bytes freed` when actually nothing was reclaimed?

Reply:
> `gc_mem_caches() > 0` would return bool aswell.
> Why is it okay that it returns `0 bytes freed` when actually nothing was reclaimed?
> I just like to understand some decisions as its probably me who needs to maintain this long-term, so please don't treat my questions as offense.

Correct, `gc_mem_caches() > 0` would return `bool` aswell. 

The point i was trying to make was that calling `gc_mem_caches()` doesn't free up memory everytime it's called.

eg. 1
```php
$storage = new Cache\Storage\Adapter\Memory();
$storage->flush();
// since no addtional memory was used. 0 bytes were freed.
```

eg. 2
```php
$storage = new Cache\Storage\Adapter\Memory();
for ($i = 0; $i <= 100000; $i++) {
    $storage->addItem('item' . $i, sha1((string) microtime(true)));
}
$storage->flush();
// since addtional memory was used. >= 1 bytes were freed.
```
**Since calling `$storage->flush()` guarantees that `$this->data` was reset `$this->data = [];` we should always return `true`.**

so i wrote the code like this.
```php
$this->data = [];
return gc_mem_caches() >= 0;
```
but it can also written like this.
```php
$this->data = [];
gc_mem_caches();

return true;
```

`unset` should immediately release memory,
the only difference between using `$this->data = null; or $this->data = [];` AND `unset($this->data);` is that the former will retain the entry in the properties store and so use more memory.

> wouldn't you need to run gc cycles first?

Yes, i've add it. You can use `var_dump(gc_status());` after `gc_collect_cycles()` is called, to generate an associative array with the following elements: `runs`, `collected`, `threshold`, and `roots`.

eg. 3 
```
// returns information about the garbage collector 
array(4) {
  ["runs"]=>
  int(5)
  ["collected"]=>
  int(100002)
  ["threshold"]=>
  int(50001)
  ["roots"]=>
  int(0)
}
```